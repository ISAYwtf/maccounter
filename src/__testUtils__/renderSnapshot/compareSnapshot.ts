import { render } from '@testing-library/react'
import React, { ReactElement } from 'react'

export const compareSnapshot = (component: ReactElement, wrapper?: React.FC) => {
    const { asFragment } = render(component, { wrapper })
    expect(asFragment()).toMatchSnapshot()
}
