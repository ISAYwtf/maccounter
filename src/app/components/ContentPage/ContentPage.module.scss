.container {
  padding: 0 0 0 var(--size-space-between-component);
  display: flex;
  //grid-gap: var(--size-space-between-component);
  //grid-template-columns: repeat(auto-fill, 420px);
  //grid-template-rows: repeat(2, 1fr);
  //grid-auto-columns: 1fr;
  //grid-auto-rows: 1fr;
  //grid-auto-flow: row dense;
  flex-wrap: wrap;
  flex-shrink: 1;
}
