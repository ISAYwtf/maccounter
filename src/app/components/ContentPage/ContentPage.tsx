import React from 'react'
import WidgetContainer from '@components/Widget/WidgetContainer'
import styles from './ContentPage.module.scss'

const ContentPage: React.FC = () => (
    <main className={styles.container}>
        <div>
            <WidgetContainer title="Last expenses" />
        </div>
    </main>
)

export default ContentPage
