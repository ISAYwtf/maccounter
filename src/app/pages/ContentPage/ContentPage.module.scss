.container {
  padding: 0 0 0 var(--size-space-between-component);
}
