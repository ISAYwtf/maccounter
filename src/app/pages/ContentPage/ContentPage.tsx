import React, { lazy, Suspense } from 'react'
import { Redirect, Route, Switch } from 'react-router-dom'
import { useSelector } from '@store/hooks'
import Home from '../Home/Home'
import styles from './ContentPage.module.scss'

const Accountancy = lazy(() => import('@pages/Accountancy/Accountancy'))
const Statistics = lazy(() => import('@pages/Statistics/Statistics'))
const Settings = lazy(() => import('@pages/Settings/Settings'))

const ContentPage: React.FC = () => {
    const links = useSelector((state) => state.app.links)
    return (
        <main className={styles.container}>
            <Suspense fallback="Загрузка...">
                <Switch>
                    <Route path={links.basedir} exact render={() => <Redirect to={links.home} />} />
                    <Route path={links.home} component={Home} />
                    <Route path={links.accountancy} component={Accountancy} />
                    <Route path={links.statistics} component={Statistics} />
                    <Route path={links.settings} component={Settings} />
                </Switch>

            </Suspense>
        </main>
    )
}

export default ContentPage
