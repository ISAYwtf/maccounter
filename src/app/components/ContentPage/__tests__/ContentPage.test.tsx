import React from 'react'
import { compareSnapshot } from '@testUtils/renderSnapshot'
import ContentPage from '@components/ContentPage/ContentPage'

describe('ContentPage', () => {
    const renderComponent = <ContentPage />

    // eslint-disable-next-line
    it('render ContentPage', () => compareSnapshot(renderComponent))
})
