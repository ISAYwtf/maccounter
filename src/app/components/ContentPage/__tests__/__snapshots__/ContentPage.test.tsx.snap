// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentPage render ContentPage 1`] = `
<DocumentFragment>
  <main
    class="container"
  >
    <div>
      <div
        class="component"
      >
        <p>
          list
        </p>
      </div>
    </div>
  </main>
</DocumentFragment>
`;
